spark/src/main/scala/org/apache/spark/sql/delta/DeltaThrowable.scala

```diff
@@ -24,8 +24,7 @@ import org.apache.spark.SparkThrowable
 trait DeltaThrowable extends SparkThrowable {
   // Portable error identifier across SQL engines
   // If null, error class or SQLSTATE is not set
-  override def getSqlState: String =
-    DeltaThrowableHelper.getSqlState(this.getErrorClass.split('.').head)
+  override def getSqlState: String = DeltaThrowableHelper.getSqlState(this.getErrorClass)
```

Comment:
This should be done anyways as it's handled by DeltaThrowableHelper in ErrorClassesJsonReader::getSqlState